src/Features/Core/Portable/MakeMethodAsynchronous/AbstractMakeMethodAsynchronousCodeFixProvider.cs

```diff
@@ -47,27 +47,22 @@ internal abstract class AbstractMakeMethodAsynchronousCodeFixProvider : CodeFixP
             }
 
             var symbol = semanticModel.GetDeclaredSymbol(node, cancellationToken) as IMethodSymbol;
+            var entryPoint = compilation.GetEntryPoint(cancellationToken);
+            bool isEntryPoint = entryPoint?.Equals(symbol) == true;
 
-            // If it's a void returning method, offer to keep the void return type, or convert to 
-            // a Task return type.
+            // Offer to convert to a Task return type.
+            context.RegisterCodeFix(
+                new MyCodeAction(GetMakeAsyncTaskFunctionResource(), c => FixNodeAsync(
+                    context.Document, diagnostic, keepVoid: false, isEntryPoint, cancellationToken: c)),
+                context.Diagnostics);
+
+            // If it's a void returning method (and not an entry point), also offer to keep the void return type
             bool isOrdinaryOrLocalFunction = symbol.IsOrdinaryMethodOrLocalFunction();
-            if (isOrdinaryOrLocalFunction && symbol.ReturnsVoid)
+            if (isOrdinaryOrLocalFunction && symbol.ReturnsVoid && !isEntryPoint)
```

Comment:
why would we gate this on isEntryPoint.  Can you have "async Task Main"?  the only rule is that it is still named Main and not MainAsync right?   it can be 'async void' or 'async Task'

Reply:
We could of course have a separate feature that would replace await with .GetAwaiter().GetResult(). I'm not sure if we should encourage people to do that though.

Reply:
That's fair.  That said, i would still not want thsi PR as is with the expensive check during analysis.  I can see about making a faster version of GetEntryPoint that would make it ok to check during analysis.

Reply:
We could just check if the method has a valid entry point signature, regardless if it's the actual entry point or not (in case the programmer specified that explicitly to the compiler for some reason).

Reply:
note: if we do this, we should write a helper that returns whether a method could be a valid entry point, so that the same could could be reused by this: #26315

Reply:
@CyrusNajmabadi Yes, the check is probably more costly than most, but it's in a fixer, not an analyzer, so I don't think it would have much impact.
Alternatively, as simple enough heuristic is to check if the method is called `Main` (and maybe check that it is public and static too). I'm happy to do that (that was my initial approach).

Reply:
@jcouv Main doesn't have to be public

Reply:
I would prefer if we created an actual helper that would check if the method is static, return type is either void, int, Task or Task<int> and it takes either no parameters or one string[] so that we could reuse this helper in other places. But I guess that can be done later, so it's up to you.

Reply:
> but it's in a fixer, not an analyzer, so I don't think it would have much impact.

Fixers have a large impact :)  both for computing if we should have a lightbulb *and* for expandin ghte lightbulb.  It's the actual *fix* that can be a bit slower (as it will have an actual dialog telling the user it's doing the update).